jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/cloudtrail/DenyPublicAccessS3BucketStoresCloudTrailLogs.java

```diff
@@ -0,0 +1,150 @@
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.NAME), s3BucketName);
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.ACCOUNTID),accountId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+        try {
+            isPublicAccessEnabled = checkForS3BucketCloudtrailLogsPublicAccess(esUrl, mustFilter);
+        } catch (Exception e) {
+            logger.error("unable to determine", e);
+            throw new RuleExecutionFailedExeption("unable to determine" + e);
+        }
+
+            if (isPublicAccessEnabled) {
+                List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+                Annotation annotation = null;
+                annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                annotation.put(PacmanSdkConstants.DESCRIPTION,"Ensure S3 bucket that stores cloudtrail Logs is not publicly accessible");
+                annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                annotation.put(PacmanRuleConstants.CATEGORY, category);
+                annotation.put(PacmanRuleConstants.NAME,resourceAttributes.get(PacmanRuleConstants.NAME));
+                issue.put(PacmanRuleConstants.VIOLATION_REASON,
+                        ruleParam.get(PacmanRuleConstants.RULE_ID) + " Violation Found!");
+                issueList.add(issue);
+                annotation.put(PacmanRuleConstants.ISSUE_DETAILS, issueList.toString());
+                logger.debug("S3 bucket that stores cloudtrail Logs is publicly accessible");
+                return new PolicyResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                        annotation);
+            }
+        }
+        logger.debug("S3 bucket that stores cloudtrail Logs is not publicly accessible");
+        logger.debug("========DenyPublicAccessS3BucketStoresCloudTrailLogs ended=========");
+        return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+
+    private boolean checkForS3BucketCloudtrailLogsPublicAccess(String esUrl, Map<String, Object> mustFilter) throws Exception {
+        logger.info("Validating the resource data from elastic search. ES URL:{}, FilterMap : {}", esUrl, mustFilter);
+        boolean validationResult = false;
+        JsonParser parser = new JsonParser();
+        JsonObject resultJson = RulesElasticSearchRepositoryUtil.getQueryDetailsFromES(esUrl, mustFilter,
+                new HashMap<>(),
+                HashMultimap.create(), null, 0, new HashMap<>(), null, null);
+
+        if (resultJson.has(PacmanRuleConstants.HITS)) {
+            String hitsString = resultJson.get(PacmanRuleConstants.HITS).toString();
+            JsonObject hitsJson = JsonParser.parseString(hitsString).getAsJsonObject();
```

Comment:
why?... `resultJson.get(PacmanRuleConstants.HITS)` is already JsonObject...

Reply:
Addressed in commit